ergo-lib/src/wallet/signing.rs

```diff
@@ -180,13 +182,51 @@ pub fn sign_transaction(
             .find(|b| b.box_id() == input.box_id)
             .ok_or(TxSigningError::InputBoxNotFound(idx))?;
         let ctx = Rc::new(make_context(state_context, &tx_context, idx)?);
+        // let hints_bag:HintsBag=tx_hints.all_hints_for_input(idx);
         prover
             .prove(
                 &input_box.ergo_tree,
                 &Env::empty(),
                 ctx,
                 message_to_sign.as_slice(),
-                &HintsBag::empty(),
+                &HintsBag::empty(), // &hints_bag,
+            )
+            .map(|proof| Input::new(input.box_id.clone(), proof.into()))
+            .map_err(|e| TxSigningError::ProverError(e, idx))
+    })?;
+    Ok(Transaction::new(
+        signed_inputs,
+        tx.data_inputs,
+        tx.output_candidates,
+    )?)
+}
+
+/// Signs a multi sig transaction (generating proofs for inputs)
+pub fn sign_transaction_multi(
```

Comment:
Is there any reason to have tx signing with hints as a separate function? I'd rather add `tx_hints: Option<TransactionHintsBag>` parameter to `sign_transaction`. The same goes for `Wallet::sign_transaction_multi` (for Wasm layer we have to keep both `sign_transaction` and `sign_transaction_multi` since there is no `Option` in JS).

Reply:
I merged the two functions even in the Wasm layer. the `TransactionHintsBag` exposed `empty` method in the Wasm layer that should be passed to `sign_transaction`; I think this resolves "multiple signers" in the Wasm layer. Is that ok or should I keep both signers?

Reply:
We as library authors should strive to keep our API easy to understand (think of newcomers). For the majority of our user's cases, they would be signing a regular tx and forcing them to learn what does `TransactionHintsBag` means and how they have to pass an `emtpy` one is an unnecessary burden. Thus, it's better to encode this optional value with types, i.e. `Option<TransactionHintsBag>`. This way, a simple glance through the docs and passing familiar `None` will be the only "price" for the majority of our users. Unfortunately, JS does not have the `Option` type, thus we need to have two `sign_transaction` and `sign_transaction_multi` in Wasm API.

Reply:
Absolutely right! I am trying in new commits to consider this point and I think now it's done.